geopandas/geodataframe.py

```diff
@@ -1582,6 +1583,9 @@ def __finalize__(self, other, method=None, **kwargs):
         # concat operation: using metadata of the first object
         elif method == "concat":
             for name in self._metadata:
+                # do not overwrite CRS if already set for the case axis=0
+                if name == "_crs" and self._crs is not None:
```

Comment:
I _think_ is no longer needed, since `_crs` is no longer a metadata name (https://github.com/geopandas/geopandas/pull/2578)